provider/google.go

```diff
@@ -322,9 +328,10 @@ func newRecord(ep *endpoint.Endpoint) *dns.ResourceRecordSet {
 	// TODO(linki): works around appending a trailing dot to TXT records. I think
 	// we should go back to storing DNS names with a trailing dot internally. This
 	// way we can use it has is here and trim it off if it exists when necessary.
-	target := ep.Target
+	targets := make([]string, len(ep.Targets))
+	copy(targets, []string(ep.Targets))
```

Comment:
I don't think there is a need to create a copy, after this stage the whole chunk of memory should be released and used no more.

Reply:
It is also worth mentioning in the comment, that the `Targets` list comes sorted

Reply:
Interestingly removing the copy causes quite some failures in tests. The tests are now complaining that expected and actual Endpoints don't match. I have to take a closer look at this.

Reply:
Is this checking a TODO? @dereulenspiegel